molr-mole-core/src/main/java/io/molr/mole/core/support/MissionPredicates.java

```diff
@@ -0,0 +1,64 @@
+package io.molr.mole.core.support;
+
+import io.molr.commons.domain.RunState;
+
+import java.util.function.Predicate;
+
+/**
+ * A general purpose validator for objects.
+ *
+ * @param <T> the type for object being validated
+ * @author HimanshuSahu31
+ */
+public class MissionPredicates<T> {
+    private Predicate<T> validator;
+
+    /**
+     * @param object an {@link Object} that represents the initial value
+     */
+    private MissionPredicates(T object) {
+        validator = o -> o.equals(object);
+    }
+
+    /**
+     * Factory method to get validator for {@link RunState}
+     *
+     * @param runState the initial value {@link RunState} for predicate function
+     * @return a {@link MissionPredicates} for validating {@link RunState}
+     */
+    public static MissionPredicates<RunState> runStatePredicate(RunState runState) {
```

Comment:
this is basically an equals, could this be named `runStateEqualsTo` ? What do you think? @HimanshuSahu31 @kaifox

Reply:
I agree. It becomes more readable.